src/UrlChecker/CakeRouterChecker.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\UrlChecker;
+
+use Cake\Routing\Exception\MissingRouteException;
+use Cake\Routing\Router;
+use InvalidArgumentException;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * Checks if a request object contains a valid URL
+ */
+class CakeRouterChecker extends DefaultUrlChecker
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check(ServerRequestInterface $request, $loginUrls, array $options = [])
+    {
+        $options = $this->_mergeDefaultOptions($options);
+        $url = $this->_getUrlFromRequest($request->getUri(), $options['checkFullUrl']);
+
+        if (!is_array($loginUrls) || empty($loginUrls)) {
```

Comment:
Default checker returns true for empty url. Should we allow for the same behavior here?

Reply:
Why does it return true? Shouldn't we always return false if it doesn't match or is empty?

Reply:
I’m not sure why. It had been that way in the FormAuthentication, before checkers were implemented.

Reply:
I'm OK with either way we just need to make a decision. :) But wouldn't it make more sense to return false if it's empty?

Reply:
It would probably be safer that way. So I’m ok with returning false on empty urls.

Reply:
Are we OK to merge this then?

Reply:
I think we need another test as I mentioned here: https://github.com/cakephp/authentication/pull/171/files#r157144178